src/MetaMorpho.sol

```diff
@@ -616,9 +616,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             uint256 supplyCap = config[id].cap;
             if (supplyCap == 0) continue;
 
-            uint256 supplyAssets = MORPHO.expectedSupplyAssets(_marketParams(id), address(this));
+            uint256 supplyShares = MORPHO.supplyShares(id, address(this));
+            (uint256 totalSupplyAssets, uint256 totalSupplyShares,,) = MORPHO.expectedMarketBalances(_marketParams(id));
 
-            totalSuppliable += supplyCap.zeroFloorSub(supplyAssets);
+            totalSuppliable += supplyCap.zeroFloorSub(supplyShares.toAssetsUp(totalSupplyAssets, totalSupplyShares));
```

Comment:
Like you have done for `_supplyMorpho` maybe would be good to also add a comment here. I know that this is more a "utility" function and not a "core" one like `_supplyMorpho` but it will be probably used very much by integrators. 

Consider also adding it as a natspec `@dev` comment.